yarn-project/noir-protocol-circuits/src/crates/types/src/utils/reader.nr

```diff
@@ -0,0 +1,40 @@
+struct Reader<N> {
```

Comment:
This is a copy from aztec-nr. I will nuke the version in aztec-nr once I implement deserialize for callstack items as well which I didn't do in this PR because I already had https://github.com/AztecProtocol/aztec-packages/pull/4381 open with a lot of changes there.